### ISSUE TITLE
Swagger does not reflect changes from the overwritten serializer

### ISSUE THREAD

**Issue by mglowinski93** [sunscrapers/djoser#688]
Hi,
I've problem with swagger not being updater after overwriting `current_user` serializer in settings.

I've modified `current_user` serializer to update  also user's  `first_name` and `last_name`, like below:

```python
class UserSerializer(serializers.UserSerializer):
    class Meta(serializers.UserSerializer.Meta):
        fields = serializers.UserSerializer.Meta.fields + (
            "first_name",
            "last_name",
        )
```
Settings:
```python
DJOSER = {
    ...
    "SERIALIZERS": {
        "current_user": "users.serializers.UserSerializer",
    },
}
```

Those changes are applied to API, but are not reflected in swagger output:
![image](https://user-images.githubusercontent.com/31914443/186605428-955462a5-e4a5-4c8b-8f68-678b7f26d367.png)

Do you have any tip how can i adjust swagger to custom serializer?

**Comment by abdelbaki1** [comment]
have you tried to replace the `user` parameter to your current serializer instead of `current-user` ?

**Comment by mglowinski93** [comment]
@abdelbaki1
That is for '/users/<user_id>' endpoint, isn't it?

And still the same problem :)

**Comment by abdelbaki1** [comment]
@mglowinski93 
you're right, i didn't pay attention to the `get_serializer_class()` method conditions.
why don 't you  override  the `retrieve()` method of `UserViewSet` class , try to adjust it according to your situation ?.
I am sorry but it 's the only i can come with it right now .

**Comment by mglowinski93** [comment]
@abdelbaki1 
Well, I did not decide to overwrite `retrieve` method in `UserViewSet`, since functionality works correctly.
There is only problem with generated swagger.
According to me, it should be generated out of the box.
Since, there are two possibilities:
1) I did something wrong
2) there is a bug
:)
